strapi/strapi
# Fix invalid openapi schema override

### What does it do?

Fixes invalid openapi schema registered as an override in the users-permissions plugin

### Why is it needed?

To have a valid openapi schema

### How to test it?

I think the best way to test it is to use https://editor.swagger.io/ to ensure the schema is valid

- First add the default config
```
openapi: 3.0.0
info:
  version: 1.0.0
  title: DOCUMENTATION
  description: ''
  termsOfService: YOUR_TERMS_OF_SERVICE_URL
  contact:
    name: TEAM
    email: contact-email@something.io
    url: mywebsite.io
  license:
    name: Apache 2.0
    url: 'https://www.apache.org/licenses/LICENSE-2.0.html'
x-strapi-config:
  path: /documentation
  showGeneratedFiles: true
  generateDefaultResponse: true
  plugins:
    - email
    - upload
    - users-permissions
servers: []
externalDocs:
  description: Find out more
  url: >-
    https://docs.strapi.io/developer-docs/latest/getting-started/introduction.html
security:
  - bearerAuth: []
```
- Then copy the content in packages/plugins/users-permissions/documentation/content-api.yaml
- Then add the following to the `components` object
```
securitySchemes:
	bearerAuth:
	  type: http
	  scheme: bearer
	  bearerFormat: JWT
Error:
  type: object
  required:
    - error
  properties:
    data:
      nullable: true
      oneOf:
        - type: object
        - type: array
          items: []
    error:
      type: object
      properties:
        status:
          type: integer
        name:
          type: string
        message:
          type: string
        details:
          type: object
```
There should be no errors in the editor

### Related issue(s)/PR(s)

resolves https://github.com/strapi/strapi/issues/13885
https://github.com/strapi/strapi/pull/14863
